testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js

```diff
@@ -5500,6 +5502,43 @@ QUnit.module('datebox validation', {}, () => {
         assert.ok(validationCallback.calledOnce, 'validationCallback called once');
         assert.strictEqual(value, '2020-10-10', 'String value passed');
     });
+
+    [false, true].forEach((isMaterial) => {
+        [{
+            validationMessagePosition: 'top',
+            checkFunction: (messagePosition, inputPosition) => messagePosition.bottom < inputPosition.top,
+        }, {
+            validationMessagePosition: 'right',
+            checkFunction: (messagePosition, inputPosition) => messagePosition.left > inputPosition.right
+        }, {
+            validationMessagePosition: 'bottom',
+            checkFunction: (messagePosition, inputPosition) => messagePosition.top > inputPosition.bottom
+        }, {
+            validationMessagePosition: 'left',
+            checkFunction: (messagePosition, inputPosition) => messagePosition.right < inputPosition.left
+        }].forEach(({ validationMessagePosition, checkFunction }) => {
+            QUnit.test(`ValidationMessage has correct position when validationMessagePosition is ${validationMessagePosition} in ${isMaterial ? 'material' : 'generic'} theme`, function(assert) {
+                const origIsMaterial = themes.isMaterial;
+                themes.isMaterial = function() { return isMaterial; };
+
+                const $dateBox = $('#dateBox').dxDateBox({
+                    type: 'date',
+                    max: new Date(1),
+                    validationMessagePosition
+                });
+                const dateBox = $dateBox.dxDateBox('instance');
+
+                dateBox.option('value', new Date());
+
+                const inputPosition = $dateBox.find(`.${TEXTEDITOR_INPUT_CLASS}`).get(0).getBoundingClientRect();
+                const messagePosition = $dateBox.find(`.${VALIDATION_MESSAGE_CONTENT}`).get(0).getBoundingClientRect();
+
+                assert.ok(checkFunction(messagePosition, inputPosition), 'position is correct');
+
+                themes.isMaterial = origIsMaterial;
```

Comment:
It seems like there is no specific code for material theme. Why should we write specific test for it?

Reply:
fair enough